Added support of object initializers

metoule opened the pull request:
This PR adds support of object initializers, so that it's possible to write `new MyClass { MyProp = 5 }`.

davideicardi commented on test/DynamicExpresso.UnitTest/ConstructorTest.cs:
```diff
@@ -50,5 +50,73 @@ public void Unknown_New_Type_Is_Not_Supported()
+		}
+
+		private class MyClass
+		{
+			public int IntField;
+			public string StrProp { get; set; }
+			public int ReadOnlyProp { get; }
+
+			public MyClass()
+            {
+            }
+
+			public MyClass(string strValue)
+			{
+				StrProp = strValue;
+			}
+
+			public override bool Equals(object obj)
+            {
+                return Equals(obj as MyClass);
+            }
+			public bool Equals(MyClass p)
+			{
+				if (p is null) return false;
```
It seems that there is some code alignment (tab/space) issue ?

davideicardi commented on test/DynamicExpresso.UnitTest/ConstructorTest.cs:
```diff
@@ -50,5 +50,73 @@ public void Unknown_New_Type_Is_Not_Supported()
+			var target = new Interpreter();
+			target.Reference(typeof(MyClass));
+
+			Assert.Throws<ParseException>(() => target.Parse("new MyClass() { StrProp }"));
```
Can you add some more tests for invalid cases, maybe like:
```
Assert.Throws<ParseException>(() => target.Parse("new MyClass() { StrProp ")); // no close bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() StrProp }")); // no open bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() {{IntField = 5}}")); // multiple bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() {5}")); // no field name
```

davideicardi reviewed (changes requested):
Great work! (thanks again ;-) )

I just added some very small comments.
If for you is fine, my plan will be to merge this PR and publish a new release that will contain several improvements thank to your work. Then in the next release we will add lambda syntax from the other PR. What do you think?

metoule commented:
I fixed the formatting issues, and added the tests you suggested.
I like your plan to release the smaller changes before the lambda syntax PR, that way there's a fallback version in case something goes terribly wrong :)

davideicardi reviewed (approved):
Great!
